aws/internal/keyvaluetags/key_value_tags.go

```diff
@@ -491,20 +491,18 @@ func (tags KeyValueTags) UrlEncode() string {
 	return values.Encode()
 }
 
-// New creates KeyValueTags from common Terraform Provider SDK types.
-// Supports map[string]string, map[string]*string, map[string]interface{}, and []interface{}.
+// New creates KeyValueTags from common types or returns an empty KeyValueTags.
+//
+// Supports various Terraform Plugin SDK types including map[string]string,
+// map[string]*string, map[string]interface{}, and []interface{}.
 // When passed []interface{}, all elements are treated as keys and assigned nil values.
+// When passed KeyValueTags or its underlying type implementation, returns itself.
 func New(i interface{}) KeyValueTags {
 	switch value := i.(type) {
+	case KeyValueTags:
+		return value
```

Comment:
`Self` or cloned-`Self`?
Is there the danger of creating an alias through which the original is unexpectedly modified?

Reply:
Probably safest to copy

Reply:
Updated in https://github.com/hashicorp/terraform-provider-aws/pull/19011/commits/c5bc89b9f3ab390a3d69ecef9211e96455103fa4 along with unit testing. 👍